arviz/stats/stats.py

```diff
@@ -1635,7 +1635,7 @@ def loo_pit(idata=None, *, y=None, y_hat=None, log_weights=None):
     ufunc_kwargs = {"n_dims": 1}
 
     if y.dtype.kind == "i" or y_hat.dtype.kind == "i":
-        y, y_hat = smooth_data(y, y_hat)
+        y, y_hat = smooth_data(y, y_hat, axis=0)
```

Comment:
I think being a bit more explicit and always reshape so that the observation is only 1d might be good, like what we are doing in `bpvplot`:

https://github.com/arviz-devs/arviz/blob/57aeded2fff88a646939fc36e5b2a86a0cf718b0/arviz/plots/backends/bokeh/bpvplot.py#L90-L94

Reply:
The approach in bpvplot has two main drawbacks, which is what I was trying to point out above, and they are still relevant there, but quite negligible, imo here they are more important. 

First one is loosing all the shape and label info when doing that. It's not clear if and/or how pointwise or group loo pits can be useful but it seems like it could be useful so I would rather not loose it for good in all cases (float data already works with 4d observations and keeps shape and labels!), i.e. hierarchical model with 3 groups, each with multiple observations, 1st group has positive bias, 2nd group has good loo pit and 3rd group has negative bias so overall loo pit looks good but group level ones do not. 

Second one is that by doing the computation via xarray, it can easily be done under the hood by numpy or by dask, so everything will still work if we have many observations so that posterior predictive array doesn't fit in memory anymore.